addons/utm/models/utm_source.py

```diff
@@ -72,12 +72,20 @@ def create(self, vals_list):
         return super().create(vals_list)
 
     def write(self, values):
+        name = values.get(self._rec_name) or values.get('name')
+        mixin = self.env['utm.mixin']
+
+        if name and len(self) > 1:
+            raise ValueError(_('Multiple records with the same name. The name should be unique!'))
         if values.get(self._rec_name) and not values.get('name'):
             values['name'] = self.env['utm.source']._generate_name(self, values[self._rec_name])
+        if name and mixin._split_name_and_count(values['name'])[0] == mixin._split_name_and_count(self[0].name)[0]:
+            values.pop('name')
+            return super().write(values)
+
```

Comment:
Hi @datonefaridze Passing by while working on a similar fix here https://github.com/odoo/enterprise/pull/60725
Can you help me understand this part?

This **if statement** is skipping value update when new name would be the same as the previous one, correct? And is doing so excluding "counts"
So when I try updating
`Justin [123]`  -> `Justin`

my update will be ignored. Feel I like am missing something in there. Can you explain that part to me?

Reply:
@nd-dew what that statement does is: it checks if name-value that you are setting is the same as already set one. for example if value written on a current mailing dato. when u try to set dato  to it, it will ignore name (before commit it used to write datoi [1] because of `_get_unique_names`).